File: engine/engine-model/src/main/scala/tools/packet/foreigneffect/ShowBuffEffect.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowBuffEffect(private var _characterId: Int, private var _skillId: Int, private var _effectId: Int,
                     private var _direction: Byte) extends PacketInput {
  def characterId: Int = _characterId

  def skillId: Int = _skillId

  def effectId: Int = _effectId

  def direction: Byte = _direction

  override def opcode(): SendOpcode = SendOpcode.SHOW_FOREIGN_EFFECT
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowBuffEffect.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowBuffEffect(Integer characterId, Integer skillId, Integer effectId,
                             Byte direction) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_FOREIGN_EFFECT;
   }
}